test/types/component-test.ts

```diff
@@ -0,0 +1,76 @@
+import * as gc from '@glimmer/component';
+import { DESTROYED, DESTROYING } from '@glimmer/component/dist/types/addon/-private/component';
+
+import { hasExactKeys } from './utils';
+
+const Component = gc.default;
+
+hasExactKeys<{
+  default: unknown,
+  setPropertyDidChange: unknown,
+  tracked: unknown,
+}>()(gc);
+
+// $ExpectType typeof Component
+gc.default;
+
+// $ExpectType () => void
+gc.setPropertyDidChange;
+
+// $ExpectType () => void
+gc.tracked;
+
+type Args = {
+  foo: number;
+};
+
+let component = new Component<Args>({}, { foo: 123 });
+
+hasExactKeys<{
+  args: unknown,
+  bounds: {
+    firstNode: unknown,
+    lastNode: unknown,
+  },
+  debugName: unknown,
+  didInsertElement: unknown,
+  didUpdate: unknown,
+  element: unknown,
+  isDestroying: unknown,
+  isDestroyed: unknown,
+  willDestroy: unknown,
+
+  // These are not public API, but technically part of the shape
```

Comment:
What is the rationale for including type tests for these?

Reply:
There's no way to _not_ include them in the type check, unfortunately, it would throw an error since they exist on the object.